modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/endpoint/EndpointMatcher.java

```diff
@@ -73,14 +72,7 @@ private String _getSingleElementRegex(APIApplication.Endpoint endpoint) {
 
 		endpointPath = endpointPath + StringPool.FORWARD_SLASH;
 
-		if (Objects.equals(
-				endpoint.getPathParameter(),
-				HeadlessBuilderConstants.PATH_PARAMETER_ID)) {
-
-			return endpointPath + "\\d+";
-		}
-
-		return endpointPath + ".+";
+		return endpointPath + "[^\\/]+$";
```

Comment:
As we are using `matches`, we don't need to add the final `$`

Reply:
make sense!